app/models/repository-contact.js

```diff
@@ -0,0 +1,33 @@
+import{ attr } from '@ember-data/model';
+import { validator, buildValidations } from 'ember-cp-validations';
+import { computed } from '@ember/object';
+import Fragment from 'ember-data-model-fragments/fragment';
+
+const Validations = buildValidations({
+  email: [
+    validator('email-format', {
+      allowBlank: true
+    })
+  ]
+});
+
+export default Fragment.extend(Validations, {
+  email: attr('string'),
+  givenName: attr('string'),
+  familyName: attr('string'),
+
+  name: computed('givenName', 'familyName', function () {
+    let name = null;
+    if (this.givenName || this.familyName) {
+      name = [this.givenName, this.familyName].join(' ');
```

Comment:
In the hypothetical scenario where this.givenName = null and this.familyName = "Foo" the name value computed will be " Foo". With the empty space before "Foo". I'm just wondering if this is the expected behaviour? If not, we may need to have additional conditional logic to deal with this.givenName being falsy or this.familyName being being falsy.

Reply:
Nice catch @wendelfabianchinsamy, we can trim the string of any white spaces.